File: src/test/java/no/nav/syfo/services/DkifServiceTest.java
package no.nav.syfo.services;

import no.nav.tjeneste.virksomhet.digitalkontaktinformasjon.v1.DigitalKontaktinformasjonV1;
import no.nav.tjeneste.virksomhet.digitalkontaktinformasjon.v1.informasjon.WSEpostadresse;
import no.nav.tjeneste.virksomhet.digitalkontaktinformasjon.v1.informasjon.WSMobiltelefonnummer;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import java.time.OffsetDateTime;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class DkifServiceTest {

    @Mock
    private DigitalKontaktinformasjonV1 dkifV1;
    @InjectMocks
    private DkifService dkifService;


    @Test
    public void verifisertSiste18mndErNullsafe() {
        boolean harVerifisert = dkifService.harVerfisertSiste18Mnd(new WSEpostadresse(), null);
        assertThat(harVerifisert).isFalse();
        harVerifisert = dkifService.harVerfisertSiste18Mnd(null, null);
        assertThat(harVerifisert).isFalse();
        harVerifisert = dkifService.harVerfisertSiste18Mnd(new WSEpostadresse(), new WSMobiltelefonnummer());
        assertThat(harVerifisert).isFalse();
    }

    @Test
    public void verifisertSiste18mnd() {
        boolean harVerifisert = dkifService.harVerfisertSiste18Mnd(new WSEpostadresse()
                .withSistVerifisert(OffsetDateTime.now().minusDays(2)), new WSMobiltelefonnummer().withSistVerifisert(OffsetDateTime.now().minusDays(4)));
        assertThat(harVerifisert).isTrue();
    }
}

File: src/test/kotlin/no/nav/syfo/kontaktinfo/consumer/DkifServiceTest.kt
package no.nav.syfo.kontaktinfo.consumer

import io.mockk.mockk
import no.nav.syfo.LocalApplication
import no.nav.syfo.util.getXMLGregorianCalendarNow
import no.nav.tjeneste.virksomhet.digitalkontaktinformasjon.v1.DigitalKontaktinformasjonV1
import no.nav.tjeneste.virksomhet.digitalkontaktinformasjon.v1.informasjon.WSEpostadresse
import no.nav.tjeneste.virksomhet.digitalkontaktinformasjon.v1.informasjon.WSMobiltelefonnummer
import org.assertj.core.api.Assertions
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.DirtiesContext
import org.springframework.test.context.junit4.SpringRunner
import javax.inject.Inject

@RunWith(SpringRunner::class)
@SpringBootTest(classes = [LocalApplication::class])
@DirtiesContext
class DkifServiceTest {
    @Inject
    private lateinit var dkifConsumer: DkifConsumer

    private val dkifV1: DigitalKontaktinformasjonV1 = mockk()

    @Test
    fun verifisertSiste18mndErNullsafe() {
        var harVerifisert = dkifConsumer.harVerfisertSiste18Mnd(WSEpostadresse(), null)
        Assertions.assertThat(harVerifisert).isFalse()
        harVerifisert = dkifConsumer.harVerfisertSiste18Mnd(null, null)
        Assertions.assertThat(harVerifisert).isFalse()
        harVerifisert = dkifConsumer.harVerfisertSiste18Mnd(WSEpostadresse(), WSMobiltelefonnummer())
        Assertions.assertThat(harVerifisert).isFalse()
    }

    @Test
    fun verifisertSiste18mnd() {
        val harVerifisert = dkifConsumer.harVerfisertSiste18Mnd(WSEpostadresse()
            .withSistVerifisert(getXMLGregorianCalendarNow(2)), WSMobiltelefonnummer().withSistVerifisert(getXMLGregorianCalendarNow(4)))
        Assertions.assertThat(harVerifisert).isTrue()
    }
}
